spec/models/job_proxy_dispatcher_spec.rb

```diff
@@ -172,5 +175,103 @@ module JobProxyDispatcherSpec
         end
       end
     end
+
+    context "with container and vms jobs" do
+      before(:each) do
+        @jobs = (@vms + @repo_vms + @container_images).collect(&:scan)
+        @dispatcher = JobProxyDispatcher.new
+      end
+
+      describe "#pending_jobs" do
+        it "returns only vm jobs by default" do
+          jobs = @dispatcher.pending_jobs
+          expect(jobs.count).to eq(@vms.count + @repo_vms.count)
+          expect(jobs.all? { |x| x.target_class == 'VmOrTemplate' }).to eq(true)
+          expect(jobs.count).to be > 0 # in case something unexpected goes wrong
+        end
+
+        it "returns only container images jobs when requested" do
+          jobs = @dispatcher.pending_jobs(ContainerImage)
+          expect(jobs.count).to eq(@container_images.count)
+          jobs.each do |x|
+            expect(x.target_class).to eq 'ContainerImage'
```

Comment:
Ah, yes, can you make the above test do what this is doing.  Feel free to say, "for followup PR". 😉